protocol/messenger_mute_test.go

```diff
@@ -107,3 +110,56 @@ func (s *MessengerMuteSuite) TestSetMute() {
 	s.Require().False(actualChat.Muted)
 	s.Require().NoError(theirMessenger.Shutdown())
 }
+
+func (s *MessengerMuteSuite) TestSetMuteForDuration() {
+	key, err := crypto.GenerateKey()
+	mockTime := 1678614777
+
+	s.Require().NoError(err)
+
+	theirMessenger, err := newMessengerWithKey(s.shh, key, s.logger, nil)
+	s.Require().NoError(err)
+
+	chatID := "status"
+
+	chat := CreatePublicChat(chatID, s.m.transport)
+
+	err = s.m.SaveChat(chat)
+	s.Require().NoError(err)
+
+	_, err = s.m.Join(chat)
+	s.Require().NoError(err)
+
+	err = theirMessenger.SaveChat(chat)
+	s.Require().NoError(err)
+
+	allChats := s.m.Chats()
+	s.Require().Len(allChats, 3)
+
+	var actualChat *Chat
+
+	for idx := range allChats {
+		if chat.ID == allChats[idx].ID {
+			actualChat = allChats[idx]
+		}
+	}
+
+	var contact *Contact
+	if actualChat.OneToOne() {
+		contact, _ = s.m.allContacts.Load(chatID)
+	}
+
+	_, error := s.m.muteChat(actualChat, contact, int64(mockTime))
+	s.NoError(error)
+	// Mock Routine
```

Comment:
This test could be improved. I was thinking about testing `watchUnmutedChats` directly. The test would basically call `watchUnmutedChats` to start polling for chats and another go routine would add dummy chats that should be muted or not, depending on the test case, and with `time.Sleep`s controlled by the test.

In order to avoid time bombs in tests, it'd be ideal if we could do time travel and control what `time.Now()` means. This is a common problem in every test suite and programming language.

Here in Go I found this [gomonkey](https://github.com/agiledragon/gomonkey) library, but maybe that's a bazooka... Anyway, it'd be good if we had a way to better control time in tests, since status-go deals a lot with such things. The simpler way is to be able to pass the current time as a parameter, but I've found so many instances where this idea can't be cleanly applied that it's good to have a second solution.

I understand this might be too much for this PR, but the idea has been shared :wink:

Reply:
I feel like it's a bazooka for such a simple task too .. But it's something to keep in mind when writing tests